### PR TITLE
Allow completion in strings and comments

### PR DESCRIPTION
**Pull request by yyoncho**
Some of the language servers do provide completion in strings and comments.

### CODE REVIEW

File: company-lsp.el [hunk 1]
```diff
@@ -451,7 +451,6 @@ See the documentation of `company-backends' for COMMAND and ARG."
     (prefix (and
              (bound-and-true-p lsp-mode)
              (lsp--capability "completionProvider")
-             (not (company-in-string-or-comment))
```
Comment (tigersoldier) [inline company-lsp.el]: I'm afraid this has to be a per-server configuration. Not all language servers handle comments and strings well.

**Comment by yyoncho** [comment]
@tigersoldier addressed your concerns - now the clients must explicitly declare that they want completion in comments. I have used nil as a default value for the newly introduced flag. Please, let me know what is your opinion on that.

**Comment by yyoncho** [comment]
ping

**Comment by yyoncho** [comment]
ping

**Comment by yyoncho** [comment]
@tigersoldier is there anything blocking merging of this PR?

**Review by tigersoldier — APPROVED** [review]

**Comment by tigersoldier** [comment]
Sorry, I somehow filtered out notifications for this PR. Merged.

**Pull request merged**